hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java

```diff
@@ -377,33 +362,48 @@ else if ( domainParamBinding.getBindType() instanceof BasicValuedMapping ) {
 					final BasicValueConverter valueConverter = jdbcMapping == null ? null : jdbcMapping.getValueConverter();
 					if ( valueConverter != null ) {
 						final Object convertedValue = valueConverter.toRelationalValue( domainParamBinding.getBindValue() );
+						if ( convertedValue != null ) {
+							for ( int i = 0; i < jdbcParamsBinds.size(); i++ ) {
+								final JdbcParametersList jdbcParams = jdbcParamsBinds.get( i );
+								assert jdbcParams.size() == 1;
+								final JdbcParameter jdbcParameter = jdbcParams.get( 0 );
+								jdbcParameterBindings.addBinding(
+										jdbcParameter,
+										new JdbcParameterBindingImpl( jdbcMapping, convertedValue )
+								);
+							}
 
+							continue;
+						}
+					}
+					final Object bindValue = domainParamBinding.getBindValue();
+					if ( bindValue == null ) {
 						for ( int i = 0; i < jdbcParamsBinds.size(); i++ ) {
 							final JdbcParametersList jdbcParams = jdbcParamsBinds.get( i );
-							assert jdbcParams.size() == 1;
-							final JdbcParameter jdbcParameter = jdbcParams.get( 0 );
-							jdbcParameterBindings.addBinding(
-									jdbcParameter,
-									new JdbcParameterBindingImpl( jdbcMapping, convertedValue )
-							);
+							for ( int j = 0; j < jdbcParams.size(); j++ ) {
+								final JdbcParameter jdbcParameter = jdbcParams.get( j );
+								jdbcParameterBindings.addBinding(
+										jdbcParameter,
+										new JdbcParameterBindingImpl( null, null )
```

Comment:
I'm actually also curious if it weren't better to pass a non-null `jdbcMapping`, even if the `bindValue` is `null`.

```suggestion
										new JdbcParameterBindingImpl( jdbcMapping, null )
```